test/accessor.test.ts

```diff
@@ -0,0 +1,448 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { test } from 'node:test';
+import assert from 'node:assert';
+import { stub, SinonStub } from 'sinon';
+
+import { IACAccessor } from '../src/accessor';
+import { HttpClient } from '@actions/http-client';
+import { VALIDATE_ENDPOINT_PATH } from '../src/commons/http_config';
+import { toBase64 } from '@google-github-actions/actions-utils';
+import { SCAN_FILE_MAX_SIZE_BYTES } from '../src/commons/constants';
+import { IACValidationException } from '../src/exception';
+
+const BASE_URL = 'base-url';
+const ORGANIZATION_ID = 'organization-id';
+const IAC = 'iac';
+const CREATE_SCAN_OPERATION_RESPONSE = {
+  name: 'operation-id',
+};
+const POLL_OPERAITON_RESPONSE = {
+  name: 'operation-id',
+  done: true,
+  response: {
+    iacValidationReport: {
+      violations: [
+        {
+          assetId: 'asset-id',
+          policyId: 'policy-id',
+          severity: 'CRITICAL',
+        },
+      ],
+    },
+  },
+};
+
+test(
+  'IACAccessor',
+  {
+    concurrency: true,
+  },
+  async (suite) => {
+    let httpClientStub: SinonStub;
+    let dateStub: SinonStub;
+    let accessor: IACAccessor;
+
+    suite.beforeEach(function () {
+      const httpClient = new HttpClient();
+      const date = new Date();
+      httpClientStub = stub(httpClient, 'request');
```

Comment:
There are examples of how to mock this node:test in deploy-cloud-functions.

Reply:
Ack.